codegen/scala-gen/src/test/scala/com/akkaserverless/codegen/scalasdk/ValueEntitySourceGeneratorSuite.scala

```diff
@@ -46,4 +46,29 @@ class ValueEntitySourceGeneratorSuite extends munit.FunSuite {
          |}
          |""".stripMargin)
   }
+  test("it can generate an abstract value entity implementation") {
+    val str = abstractEntity(testData.valueEntity(domainParent), testData.simpleEntityService(apiParent))
+    assertNoDiff(
+      str,
+      s"""package com.example.service.domain
+          |
+          |import com.akkaserverless.scalasdk.valueentity.ValueEntity
+          |import com.akkaserverless.scalasdk.valueentity.ValueEntity.Effect
+          |import com.example.service.GetValue
+          |import com.example.service.MyState
+          |import com.example.service.SetValue
+          |import com.external.Empty
+          |
+          |/** A value entity. */
+          |abstract class AbstractMyValueEntity extends ValueEntity[MyState] {
+          |
+          |  /** Command handler for "Set". */
+          |  def set(currentState: MyState, setValue: SetValue): Effect[Empty]
+          |
+          |  /** Command handler for "Get". */
+          |  def get(currentState: MyState, getValue: GetValue): Effect[MyState]
```

Comment:
these should be `MyState` classes from 2 different classes, right? Check out `generateImplementationSkeleton` for how it was done there

Reply:
Thanks, will have a look.